mllib/src/main/scala/org/apache/spark/mllib/pmml/export/TreeModelUtils.scala

```diff
@@ -0,0 +1,275 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.mllib.pmml.export
+
+import scala.annotation.tailrec
+import scala.collection.JavaConverters._
+import scala.collection.mutable.MutableList
+
+import org.dmg.pmml.{Node => PMMLNode, _}
+
+import org.apache.spark.mllib.tree.configuration.{Algo, FeatureType}
+import org.apache.spark.mllib.tree.configuration.Algo._
+import org.apache.spark.mllib.tree.model.{DecisionTreeModel, Node}
+
+private[mllib] object TreeModelUtils {
+
+
+  def toPMMLTree(mllibTreeModel: DecisionTreeModel, modelName: String): TreeModel = {
+
+    val miningFunctionType = getPMMLMiningFunctionType(mllibTreeModel.algo)
+
+    val miningSchema = new MiningSchema()
+      .withMiningFields(getMiningFieldsForTree(mllibTreeModel).asJava)
+
+    val treeModel = new org.dmg.pmml.TreeModel()
+      .withModelName(modelName)
+      .withFunctionName(miningFunctionType)
+      .withNode(buildStub(mllibTreeModel.topNode))
+      .withMiningSchema(miningSchema)
+
+    treeModel
+
+  }
+
+  /** Build a pmml tree stub given the root mllib node. */
+  private def buildStub(rootNode: Node): PMMLNode = {
+    // get rootPMML node for the MLLib node
+    val rootPMMLNode = createNode(rootNode)
+
+    if (!rootNode.isLeaf) {
+      // if left node exist, add the node
+      if (rootNode.leftNode.isDefined) {
+        val leftNode = buildStub(rootNode.leftNode.get)
+        rootPMMLNode.withNodes(leftNode)
+      }
+      // if right node exist, add the node
+      if (rootNode.rightNode.isDefined) {
+        val rightNode = buildStub(rootNode.rightNode.get)
+        rootPMMLNode.withNodes(rightNode)
+      }
+    }
+
+    rootPMMLNode
+  }
+
+  /** Get pmml Predicate for a given mlLib tree node. */
+  private def getPredicate(node: Node): Option[Predicate] = {
+    // compound predicate if classification and categories list length > 0
+
+    if (node.split.isDefined) {
+      val split = node.split.get
+      val featureType = split.featureType
+      val fieldName = FieldName.create("field_" + split.feature.toString)
```

Comment:
Here you are recreating the Field Name, once again, couldn't you get from the DataFields?